### PR TITLE
get sum and related tests

### PR DESCRIPTION
**Pull request by lohanidamodar**
Adds new method to get sum of an attribute

### CODE REVIEW

File: tests/Database/Base.php [hunk 1]
```diff
@@ -982,6 +983,32 @@ public function testCount()
         Authorization::reset();
     }
 
+    /**
+     * @depends testFind
+     */
+    public function testSum()
```
Comment (kodumbeats) [inline tests/Database/Base.php:989]: Can we also test the `$max` param?

File: src/Database/Database.php [hunk 2]
```diff
@@ -1049,6 +1049,25 @@ public function count(string $collection, array $queries = [], int $max = 0): in
...
+     * @param string $collection
+     * @param string $attribute
+     * @param Query[] $queries
+     * @param int $max
+     *
+     * @return int|float
+     */
+    public function sum(string $collection, string $attribute, array $queries = [], int $max = 0)
+    {
+        $count = $this->adapter->sum($collection, $attribute, $queries, $max);
```
Comment (kodumbeats) [inline src/Database/Database.php:1066]: If we're enforcing a `$max` limit, do we also need to worry about orders/offsets when calling `sum()`?

File: src/Database/Adapter/MongoDB.php [hunk 3]
```diff
@@ -538,6 +537,49 @@ public function count(string $collection, array $queries = [], int $max = 0): in
...
+        if(!empty($queries)) {
+            $pipeline[] = ['$match' => $filters];
+        }
+        $pipeline[] = [
+                '$group' => [
+                    '_id' => null,
+                    'total' => ['$sum' => '$' . $attribute],
+                ],
+        ];
+        return ($collection->aggregate($pipeline, $options)->toArray()[0] ?? [])['total'] ?? 0;
```
Comment (kodumbeats) [inline src/Database/Adapter/MongoDB.php:580]: Can we get some documentation here? After `$collection->aggregate()`, I'm not sure what's going on.

**Review by kodumbeats — CHANGES_REQUESTED** [review]
:)

File: src/Database/Adapter/MongoDB.php (hunk 3 above)
Comment (lohanidamodar) [reply src/Database/Adapter/MongoDB.php:580]: Not even me, it took quite a bit of Google search for both @TorstenDittmann and met to get this working. 

Will add what I understood.

File: src/Database/Database.php (hunk 2 above)
Comment (lohanidamodar) [reply src/Database/Database.php:1066]: @kodumbeats I'm not sure either, I basically replicated `count` method, and only changed the SQL query to get the `sum`. So It should work exactly as `count` works with `$max`, WDYT?

File: tests/Database/Base.php (hunk 1 above)
Comment (lohanidamodar) [reply tests/Database/Base.php:988]: sure

File: src/Database/Adapter.php [hunk 4]
```diff
@@ -243,6 +243,18 @@ abstract public function find(string $collection, array $queries = [], int $limi
...
+     * @param string $attribute
+     * @param Query[] $queries
+     * @param int $max
+     *
+     * @return int|float
+     */
+    abstract public function sum(string $collection, string $attribute, array $queries = [], int $max = 0);
+
+    /**
+     * Sum an attribute
```
[the remark this answers is not in the document]
Comment (lohanidamodar) [reply src/Database/Adapter.php:255]: It was my mistake.

**Review by eldadfux — APPROVED** [review]
Nothing to add on top of @kodumbeats comments.

File: src/Database/Adapter/MongoDB.php (hunk 3 above)
Comment (lohanidamodar) [reply src/Database/Adapter/MongoDB.php]: Updated with comments

**Comment by lohanidamodar** [comment]
> Nothing to add on top of @kodumbeats comments.

Fixed, mongodb sum and updated the tests and comments as requested.

File: tests/Database/Base.php [hunk 5]
```diff
@@ -996,6 +996,9 @@ public function testSum()
         $this->assertEquals(round(39.50+25.99, 2), round($count, 2));
         $count = static::getDatabase()->sum('movies', 'price', [new Query('year', Query::TYPE_EQUAL, [2019]),]);
         $this->assertEquals(round(39.50+25.99, 2), round($count, 2));
+        
+        $count = static::getDatabase()->sum('movies', 'year', [new Query('year', Query::TYPE_EQUAL, [2019])], 1);
```
Comment (eldadfux) [inline tests/Database/Base.php:1000]:
```suggestion
        $sum = static::getDatabase()->sum('movies', 'year', [new Query('year', Query::TYPE_EQUAL, [2019])], 1);
```

File: tests/Database/Base.php [hunk 6]
```diff
@@ -996,6 +996,9 @@ public function testSum()
...
+        $this->assertEquals(2019, $count);
```
Comment (eldadfux) [inline tests/Database/Base.php:1001]:
```suggestion
        $this->assertEquals(2019, $sum);
```

**Review by eldadfux — CHANGES_REQUESTED** [review]

**Comment by lohanidamodar** [comment]
@eldadfux updated fixed the naming issue in the tests

**Review by eldadfux — APPROVED** [review]

File: src/Database/Database.php (hunk 2 above)
Comment (kodumbeats) [reply src/Database/Database.php:1066]: Fair enough - if we need that functionality, we can add it later 👍🏻

**Review by kodumbeats — APPROVED** [review]

**Pull request merged**